proxmark3-rrg: Initial commit

nyanotech opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Initial commit of the [rfidresearchgroup fork](https://github.com/RfidResearchGroup/proxmark3) of proxmark3. 

Also, add myself to `maintainer-list.nix`

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -6653,6 +6653,12 @@
     githubId = 7677321;
     name = "Paul Trehiou";
   };
+  nyanotech = {
```
Please put this into a separate commit with the message ``maintainers: add nyanotech``.

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+
+, qtSupport ? true }:
+
+with lib;
```
Please use with lib where needed.

```suggestion
```

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+{ lib, mkDerivation, stdenv, fetchFromGitHub, pkgconfig, gcc-arm-embedded
+, bluez5, qt5, readline
+
+, hardwarePlatform ? "PM3RDV4"
+
+, hardwarePlatformExtras ? ""
+
+, bluetoothSupport ? true
+
+, qtSupport ? true }:
```
```suggestion
, hardwarePlatform ? "PM3RDV4"
, hardwarePlatformExtras ? ""
, bluetoothSupport ? true
, qtSupport ? true }:
```

Is there a use case when we want to disable qt or bluetooth? If not please remove those options.

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+{ lib, mkDerivation, stdenv, fetchFromGitHub, pkgconfig, gcc-arm-embedded
```
```suggestion
{ stdenv, lib, mkDerivation, fetchFromGitHub, pkg-config, gcc-arm-embedded
```

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+    sha256 = "13xrhvrsm73rfgqpgca6a37c3jixdkxvfggmacnnx5fdfb393bfx";
+  };
+
+  nativeBuildInputs = [ pkgconfig gcc-arm-embedded ];
```
```suggestion
  nativeBuildInputs = [ pkg-config gcc-arm-embedded ];
```

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+{ lib, mkDerivation, stdenv, fetchFromGitHub, pkgconfig, gcc-arm-embedded
+, bluez5, qt5, readline
```
qt5 should not be inputted directly but through libsForQt5.callPackage in top level.

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+  nativeBuildInputs = [ pkgconfig gcc-arm-embedded ];
+  buildInputs = [ readline ]
+    ++ optional bluetoothSupport bluez5
+    ++ optional qtSupport qt5.qtbase;
```
```suggestion
    ++ optional qtSupport qtbase;
```

SuperSandro2000 commented on pkgs/tools/security/proxmark3/proxmark3-rrg.nix:
```diff
@@ -0,0 +1,47 @@
+    ++ optional qtSupport qt5.qtbase;
+
+  buildPhase = ''
+    make PLATFORM=${hardwarePlatform} \
```
Please use makeFlags instead.

nyanotech committed:
maintainers: Add nyanotech

nyanotech committed:
proxmark3-rrg: Initial commit

The rfidresearchgroup fork is a significant fork from the upstream, and
the build steps are more than a bit different, which is why I did this
in a new file.

Also, add myself to the maintainers list
